### DISCUSSION TITLE
Request: security updates for 1.15

### DISCUSSION THREAD

**Discussion by jamiemccarthy** [sparklemotion/nokogiri#3146] [RFCs]
Yes, ruby 2.7 is EOL for almost a year, but many of us still have blockers before we can upgrade to 3.0. As of right now, CVE-2024-25062 is one of our top two security concerns on an audience-facing site I work on. Our current path forward would be figuring out how to compile libxml2 and link it in to nokogiri for ourselves, which is going to be tricky to do right for our multi-platform Docker-based environments.

If you're able to continue to release (only) security fixes for the 1.15 version tree, we and probably many other sites would appreciate it.

Thank you for your consideration, and for your years of work on this important and popular gem!

**Comment by flavorjones** [comment]
Hi @jamiemccarthy, sorry for my slow reply, it's been a busy week.

I'll take a look at how hard this will be. The easiest path would be to cut a v1.15 patch release that upgrades to [libxml 2.11.7](https://gitlab.gnome.org/GNOME/libxml2/-/releases/v2.11.7) which patches this CVE. If that goes green I'm open to cutting a security release.

**Reply by flavorjones** [reply]
Initial signs are good, the PR is up at https://github.com/sparklemotion/nokogiri/pull/3154

**Reply by flavorjones** [reply]
While that's running through CI, I'll just note that after release I will also have to update the GHSA and the matching entry in ruby-advisory-db, so that `bundle audit`, `dependabot`, and other tooling will give people correct guidance.

**Reply by flavorjones** [reply]
[Release 1.15.6 / 2024-03-16 · sparklemotion/nokogiri](https://github.com/sparklemotion/nokogiri/releases/tag/v1.15.6)

**Reply by flavorjones** [reply]
Updated GHSA: https://github.com/sparklemotion/nokogiri/security/advisories/GHSA-xc9x-jj77-9p9j

**Reply by jamiemccarthy** [reply]
Thanks so much for doing this, I really appreciate it. You rock 🎉

**Reply by flavorjones** [reply]
:bow: you're welcome!
